samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -73,17 +61,17 @@
   private final Config config;
   private final boolean fetchThresholdBytesEnabled;
   private final KafkaSystemConsumerMetrics metrics;
-  private final KafkaStartpointRegistrationHandler kafkaStartpointRegistrationHandler;
 
   // This sink is used to transfer the messages from the proxy/consumer to the BlockingEnvelopeMap.
   final KafkaConsumerMessageSink messageSink;
 
   // This proxy contains a separate thread, which reads kafka messages (with consumer.poll()) and populates
   // BlockingEnvelopMap's buffers.
-  private final KafkaConsumerProxy proxy;
+  private final KafkaConsumerProxy<K, V> proxy;
 
-  // Holds mapping from {@code TopicPartition} to registered {@code Startpoint}. This will be used in the start().
-  final Map<TopicPartition, Startpoint> topicPartitionToStartpointMap = new HashMap<>();
+  // keep registration data until the start - mapping between registered SSPs and topicPartitions, and their offsets
+  Map<TopicPartition, String> topicPartitionsToOffset = new HashMap<>();
```

Comment:
Instead of having two parallel maps, could you have a single map from `TopicPartition` to "ssp with offset"? It will help ensure they don't get out of sync.